cmd/otelcol/main_windows.go

```diff
@@ -18,25 +18,35 @@ package main
 
 import (
 	"fmt"
+	"os"
 
 	"golang.org/x/sys/windows/svc"
 
 	"go.opentelemetry.io/collector/service"
 )
 
 func run(params service.Parameters) error {
-	isInteractive, err := svc.IsAnInteractiveSession()
-	if err != nil {
-		return fmt.Errorf("failed to determine if we are running in an interactive session %w", err)
-	}
-
-	if isInteractive {
+	if useInteractiveMode, err := checkUseInteractiveMode(); err != nil {
+		return err
+	} else if useInteractiveMode {
 		return runInteractive(params)
 	} else {
 		return runService(params)
 	}
 }
 
+func checkUseInteractiveMode() (bool, error) {
+	if value, present := os.LookupEnv("NO_WINDOWS_SERVICE"); present && value != "0" {
```

Comment:
Add a comment to explain what this does.

Reply:
Added.